### PR TITLE
[Docs] Docs customization

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | docs for #12693 
| License         | MIT

### CODE REVIEW

File: docs/customization/api.rst [hunk 1]
```diff
@@ -532,6 +532,50 @@ And here we go, now your response should look like this:
         //...
     }
 
+Configuring endpoints using yaml
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (CoderMaggie) [inline docs/customization/api.rst:536]:
```suggestion
Configuring endpoints using yaml
--------------------------------
```

File: docs/customization/api.rst [hunk 2]
```diff
@@ -532,6 +532,50 @@ And here we go, now your response should look like this:
...
+
+To remove endpoint from api all you need to do is specify which resource and operation
+should be removed in ``config/api_platform/config.yaml``.
```
Comment (CoderMaggie) [inline docs/customization/api.rst:539]:
```suggestion
To remove an endpoint from the API using YAML you need to specify the operation of which resource
should be removed in ``config/api_platform/config.yaml``.
```

File: docs/customization/api.rst [hunk 3]
```diff
@@ -532,6 +532,50 @@ And here we go, now your response should look like this:
...
+
+If you want to remove for example admin endpoint ``GET`` from ``Zones`` you need to use ``(unset)`` key in yaml config.
```
Comment (CoderMaggie) [inline docs/customization/api.rst:541]:
```suggestion
If you want to remove, for example, the admin ``GET`` endpoint of ``Zones``, you need to use the ``(unset)`` key in its yaml config.
```

File: docs/customization/api.rst [hunk 4]
```diff
@@ -532,6 +532,50 @@ And here we go, now your response should look like this:
...
+
+.. code-block:: yaml
+    '%sylius.model.zone.class%':
+        collectionOperations:
+            admin_get (unset): ~
+
+
+Using the ``(unset)`` key you can remove filters you dont need.
```
Comment (CoderMaggie) [inline docs/customization/api.rst:549]:
```suggestion
Using the ``(unset)`` key you can also remove filters you don't need.
```

File: docs/customization/api.rst [hunk 5]
```diff
@@ -532,6 +532,50 @@ And here we go, now your response should look like this:
...
+
+.. code-block:: yaml
+    '%sylius.model.product.class%':
+        collectionOperations:
+            shop_get:
+                filters (unset):
+                    - app.product_new_filter
+
+If you need to add new filter in yaml configuration, simply place this code into ``config/api_platform/config.yaml``
```
Comment (CoderMaggie) [inline docs/customization/api.rst:558]:
```suggestion
If you need to add a new filter in the yaml configuration, simply add this kind of code to ``config/api_platform/config.yaml``
```

File: docs/customization/api.rst [hunk 6]
```diff
@@ -532,6 +532,50 @@ And here we go, now your response should look like this:
...
+
+.. code-block:: yaml
+    '%sylius.model.product.class%':
+        collectionOperations:
+            shop_get:
+                filters:
+                    - app.product_new_filter
+
+
+To add new operation all you need to do is specify it in config file.
```
Comment (CoderMaggie) [inline docs/customization/api.rst:568]:
```suggestion
To add a new operation, just specify it in the config file.
```

**Review by lchrusciel — APPROVED** [review]

File: docs/customization/api.rst [hunk 7]
```diff
@@ -532,6 +532,54 @@ And here we go, now your response should look like this:
...
+.. code-block:: yaml
+
+    '%sylius.model.channel.class%':
+        collectionOperations
+            my_new_operation:
+                method: GET
+                path: /shop/channels
+                normalization_context:
+                    groups: ['shop:channel:read']
+
```
Comment (AdamKasp) [inline docs/customization/api.rst:582]: IIRC you can also add a case with overwriting the entire endpoint configuration (you firstly should unset operation and after that add new one).

**Review by AdamKasp — APPROVED** [review]

File: docs/customization/api.rst (hunk 7 above)
Comment (SirDomin) [reply docs/customization/api.rst:582]: good point, added a way to overwrite existing endpoint (we dont need to unset it)

**Review by Tomanhez — APPROVED** [review]

File: docs/customization/api.rst [hunk 8]
```diff
@@ -532,6 +532,66 @@ And here we go, now your response should look like this:
...
+.. code-block:: yaml
+
+    '%sylius.model.order.class%':
+        collectionOperations:
+            admin_get:
+                path: /admin/orders/new_endpoint
+                normalization_context:
+                    groups: ['shop:channel:new_group']
+
+This way we can edit existing endpoint and add custom normalization or change path.
```
Comment (GSadee) [inline docs/customization/api.rst:594]:
```suggestion
This way we can edit the existing endpoint and add custom normalization or change the path.

```

Comment (GSadee) [inline docs/customization/api.rst:594]:
```suggestion
This way we can edit the existing endpoint and add custom normalization or change the path.

```

**Review by GSadee — APPROVED** [review]

**Review by arti0090 — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, @SirDomin! :tada:

**Pull request merged**